wgpu_shadertoy/inputs.py

```diff
@@ -97,7 +216,54 @@ class ShadertoyChannelSoundcloud(ShadertoyChannel):
 
 
 class ShadertoyChannelBuffer(ShadertoyChannel):
-    pass
+    """
+    Shadertoy buffer texture input. The relevant code and renderpass resides in the main Shadertoy class.
+    Parameters:
+        buffer (str|`BufferRenderPass`): The buffer index, can be one of ("A", "B", "C", "D"), or the `BufferRenderPass` itself.
+        parent (`RenderPass`): The main renderpass this buffer-texture is attached to. (optional in the init, but should be set later).
+        **kwargs for sampler settings.
+    """
+
+    def __init__(self, buffer, parent=None, **kwargs):
+        super().__init__(**kwargs)
+        if isinstance(buffer, str):
+            self.buffer_idx = buffer.lower()  # A,B,C or D?
+            self._renderpass = None
+        else:
+            # TODO can we check for instance of BufferRenderPass?
+            self._renderpass = buffer
+            self.buffer_idx = buffer.buffer_idx
+
+        if parent is not None:
+            self._parent = parent
+        self.dynamic = True
+
+    @property
+    def size(self):
+        # width, height, 1, ?
+        texture_size = self.renderpass.texture_size
+        return (texture_size[1], texture_size[0], 1)
+
+    @property
+    def renderpass(self):  # -> BufferRenderPass:
+        if self._renderpass is None:
+            self._renderpass = self.parent.main.buffers[self.buffer_idx]
```

Comment:
this causes issues when the channel buffer class references a `buffer_idx` that has not been given an actual `BufferPass` for the main class. This can happen via the API (example: https://www.shadertoy.com/view/4XcXWM) too.
non existing Buffers should probably sample the black texture, but we can't replace this class in place - need to detect this earlier